spec/features/author_management/author_vs_artist_labels_spec.rb

```diff
@@ -8,40 +8,86 @@
 # and deleting authors and artists.
 describe 'Author and Artist labels', :feature, js: true do
   let(:submitter) { FactoryBot.create(:submitter) }
+  let(:book) { FactoryBot.create(:book, author_first_name: ['First0'], author_last_name: ['Last0']) }
+  let(:book_id) { book.id }
+  let(:artwork) { FactoryBot.create(:artwork, author_first_name: ['First0'], author_last_name: ['Last0']) }
+  let(:artwork_id) { artwork.id }
 
   before do
     create_submitter(submitter)
   end
 
   it 'uses the title of Author for new books' do
     visit new_book_path
+    expect(page).to have_selector("input[name='book[author_first_name][]']", visible: true)
+
     expect(page).to have_content('Add Author')
     expect(page).not_to have_content('Add Artist')
-    first_name_fields.last.set('First0')
-    last_name_fields.last.set('Last0')
-    click_on 'Add Author'
-    first_name_fields.last.set('First1')
-    last_name_fields.last.set('Last1')
-    click_on 'Add Author'
+
+    add_author_or_artist_and_verify_field
+    add_author_or_artist_and_verify_field
+
     expect(page).to have_selector('button', text: 'Remove Author', count: 2)
     expect(page).not_to have_selector('button', text: 'Remove Artist')
+
     first('button', text: 'Remove Author').click
     expect(page).to have_selector('button', text: 'Remove Author', count: 1)
     expect(page).not_to have_content('Artist')
   end
 
   it 'has the title of Artist for new artworks' do
     visit new_artwork_path
+    expect(page).to have_selector("input[name='artwork[author_first_name][]']", visible: true)
+
+    expect(page).to have_content('Add Artist')
+    expect(page).not_to have_content('Add Author')
+
+    add_author_or_artist_and_verify_field
+    add_author_or_artist_and_verify_field
+
+    expect(page).to have_selector('button', text: 'Remove Artist', count: 2)
+    expect(page).not_to have_selector('button', text: 'Remove Author')
+
+    first('button', text: 'Remove Artist').click
+    expect(page).to have_selector('button', text: 'Remove Artist', count: 1)
+    expect(page).not_to have_content('Author')
+  end
+
+  it 'uses the title of Author for existing books' do
+    login_as_admin
+
+    visit edit_book_path(book)
+    expect(page).to have_selector("input[name='book[author_first_name][]']", visible: true)
+
+    expect(page).to have_content('Add Author')
+    expect(page).not_to have_content('Add Artist')
```

Comment:
@Janell-Huyck You still have these checks for 'Add Author' and 'Add Artist' in several places in this file even though your new `add_author_or_artist_and_verify_field` method also checks for them.  Can you safely get rid of these expects?

Reply:
Part of what I learned with this pr is that the statement `expect(page).to have_selector("input[name='book[author_first_name][]']"` has the side-effect of causing Capybara to wait until these elements are loaded before proceeding.  

The problems I was having have to do with Capybara proceeding to check things before pages have actually loaded.  The best way I found to slow Capybara down is to add these checks in before I check what I'm actually needing to check.

Are you aware of another way to do this?  I have tried to sprinkle this check in anywhere where Capybara might have been expected to either initially load or to reload a page due to some server-side Javascript actions.  It will wait up to the `default_wait_time` for these things to load, but if they're there it proceeds immediately.

Reply:
I was only referring to lines 62 & 63.  They verify the "Add Author" or "Add Artist" buttons are there, but then right after that you call `add_author_or_artist_and_verify_field`, which does its own (second) check that "Add Author" or "Add Artist" are present.  

So I'm wondering if `expect(page).to have_content('Add Author')` and `expect(page).not_to have_content('Add Artist')` are redundant because of `add_author_or_artist_and_verify_field`?  Or are they needed to slow Capybara down?

Reply:
Ah, I see what you're saying.  The helper method is checking that there's only one type of button, but it doesn't really care which one is there.  This test here is more about making that the correct button type is present.  The "book" model should have "author", and if it had "artist" that should fail.  The add_author_or_artist_and_verify_field method will pass for either one.

Reply:
Rebasing... hang on

Reply:
Thanks for the explanation and the rebase.  I'll merge when the build goes green. (might want to look into turning on parallel testing in CI)